### PR TITLE
Refactor Lung Handling

### PR DESCRIPTION
**Pull request by Fox-McCloud**
After glancing at lungs, I noticed there was a mistake in how things were handled; it still had a list for checking gas requirements, yet everything was based on vars......upon glancing over it further, I was a bit disappointed that not everything was customizeable, sooooo, here we are.

This basically changes lungs so you can customize a lot about them:

- Nitrogen is a valid breathing option
- Can specify what type of damage you take if you breathe in a gas
  - Not getting enough of something is still going to be respiration (oxy) damage
- Can specify what heat and cold level thresholds are needed before you start taking damage
- Can specify the type of damage you take when the cold/heat levels are too much
- Can specify the amount of damage you take from breathing cold/hot air
- Can specify which reagent will stabilize you in crit

Code changes
- alerts are consistently named now
- added brain damage to various damage dealing procs

Changes
- Breathing tox causes tox damage directly
- Species hotmod/coldmod now impacts the damage taken from breathign hot/cold air
- You get warnings, once again, when breathing hot/cold air that damages you

:cl: Fox McCloud
tweak: breathing plasma now causes direct tox damage
tweak: breathing hot/cold air now warns you when you're doing so, again
tweak: species heat/cold mod now impacts damage from breathing hot/cold gases
/:cl:

### CODE REVIEW

File: code/modules/surgery/organs/lungs.dm [hunk 1]
```diff
@@ -15,13 +15,13 @@
 	slot = "lungs"
 	gender = PLURAL
 	w_class = WEIGHT_CLASS_NORMAL
-	var/list/breathlevels = list("safe_oxygen_min" = 16,"safe_oxygen_max" = 0,"safe_co2_min" = 0,"safe_co2_max" = 10,
-	"safe_toxins_min" = 0,"safe_toxins_max" = 0.05,"SA_para_min" = 1,"SA_sleep_min" = 5,"BZ_trip_balls_min" = 1)
```
Comment (Fox-McCloud) [inline code/modules/surgery/organs/lungs.dm]: I have a feeling this was a mistake; it was only `safe_co2_min` that uses it.

**Comment by banthebantz** [comment]
brain damage for asphyxiation WHEN

**Comment by meeeeme** [comment]
Smoking should have a 1% chance of giving you lung cancer

File: code/_onclick/hud/alert.dm [hunk 2]
```diff
@@ -137,8 +146,7 @@ The box in your backpack has an oxygen tank and breath mask in it."
 
 /obj/screen/alert/tox_in_air
```
Comment (nicbn) [inline code/_onclick/hud/alert.dm]: Rename to too_much_tox

File: code/_onclick/hud/alert.dm [hunk 3]
```diff
@@ -137,8 +146,7 @@ The box in your backpack has an oxygen tank and breath mask in it."
...
 	name = "Choking (Plasma)"
-	desc = "There's highly flammable, toxic plasma in the air and you're breathing it in. Find some fresh air. \
-The box in your backpack has an oxygen tank and gas mask in it."
+	desc = "There's highly flammable, toxic plasma in the air and you're breathing it in. Find some fresh air. The box in your backpack has an oxygen tank and gas mask in it."
 	icon_state = "tox_in_air"
```
Comment (nicbn) [inline code/_onclick/hud/alert.dm]: Same for here and for the dmi.

File: code/_onclick/hud/alert.dm (hunk 2 above)
Comment (Fox-McCloud) [inline code/_onclick/hud/alert.dm]: Yup.

File: code/modules/surgery/organs/lungs.dm [hunk 4]
```diff
@@ -228,21 +278,21 @@
```
Comment (Fox-McCloud) [inline code/modules/surgery/organs/lungs.dm]: Wonder if I should remove line 273: `if(abs(310.15 - breath.temperature) > 50)` for even more variety of lungs...

File: code/modules/surgery/organs/lungs.dm [hunk 5]
```diff
@@ -48,30 +70,33 @@
...
+		H.failed_last_breath = TRUE
 		if(safe_oxygen_min)
-			H.throw_alert("oxy", /obj/screen/alert/oxy)
+			H.throw_alert("not_enough_oxy", /obj/screen/alert/not_enough_oxy)
 		else if(safe_toxins_min)
 			H.throw_alert("not_enough_tox", /obj/screen/alert/not_enough_tox)
 		else if(safe_co2_min)
 			H.throw_alert("not_enough_co2", /obj/screen/alert/not_enough_co2)
-		return 0
+		else if (safe_nitro_min)
```
Comment (Fox-McCloud) [inline code/modules/surgery/organs/lungs.dm]: no space.

File: code/modules/surgery/organs/lungs.dm [hunk 6]
```diff
@@ -228,21 +278,21 @@
 
 		if(!(GLOB.mutations_list[COLDRES] in H.dna.mutations) && !(RESISTCOLD in species_traits)) // COLD DAMAGE
 			switch(breath.temperature)
```
Comment (Fox-McCloud) [inline code/modules/surgery/organs/lungs.dm]: It's rather peculiar, just by making the below changes, you only ever take heat damage, even when breathing in something too cold. Can't figure this one out. Any ideas @AnturK @Cyberboss

Comment (AnturK) [inline code/modules/surgery/organs/lungs.dm]: Switch requires constants.

**Comment by MoreRobustThanYou** [comment]
> Nitrogen is a valid breathing option

we all know what comes next

**Comment by Fox-McCloud** [comment]
>we all know what comes next

NO.

File: code/modules/surgery/organs/lungs.dm (hunk 6 above)
Comment (Fox-McCloud) [inline code/modules/surgery/organs/lungs.dm]: I feel like an idiot now. Either case, thanks for the info.

File: code/modules/surgery/organs/lungs.dm [hunk 7]
```diff
@@ -199,50 +251,56 @@
...
+		var/heat_modifier = H.dna.species.heatmod
+		if(breath_temperature > heat_level_1_threshold && breath_temperature < heat_level_2_threshold)
+			H.apply_damage_type(heat_level_1_damage*heat_modifier, heat_damage_type)
+		if(breath_temperature > heat_level_2_threshold && breath_temperature < heat_level_3_threshold)
+			H.apply_damage_type(heat_level_2_damage*heat_modifier, heat_damage_type)
+		if(breath_temperature > heat_level_3_threshold)
+			H.apply_damage_type(heat_level_3_damage*heat_modifier, heat_damage_type)
+		if(breath_temperature > heat_level_1_threshold)
+			if(prob(20))
+				to_chat(H, "<span class='warning'>You feel [hot_message] in your [name]!</span>")
```
Comment (Fox-McCloud) [inline code/modules/surgery/organs/lungs.dm]: Because lungs are plural, this has to be `name` as opposed to `src`.

**Comment by Fox-McCloud** [comment]
@AnturK 

Tested this out; everything functions as intended now; I went ahead changed a couple of minor things:

- Species heatmod/coldmod impacts how much damage breathing overly hot/cold air causes
- Breathing plasma causes tox damage directly (so we can make it so breathing in plasma could deal other damage types)
- Re-add the warning for breathing in overly hot/cold air

File: code/modules/surgery/organs/lungs.dm (hunk 1 above)
Comment (lzimann) [inline code/modules/surgery/organs/lungs.dm]: Maybe a leftover from something else.

**Review by Cyberboss — APPROVED** [review]

**Comment by MoreRobustThanYou** [comment]
NEEXXXTTTT we have someone trying to add Vox!

**Pull request merged**